runelite-client/src/main/java/net/runelite/client/plugins/customcursor/CustomCursorPlugin.java

```diff
@@ -110,8 +129,36 @@ private void updateCursor()
 			}
 			return;
 		}
+		else if (selectedCursor == CustomCursor.EQUIPPED_WEAPON)
```

Comment:
Shouldn't this be on client thread?

Reply:
Can I get a pointer about what I need to do to fix this? Sorry to bug, I'm new to this project so I must be missing something.

Reply:
The client itself runs on a specific thread, and you can queue operations which invoke client functions using our `ClientThread` class. Search the codebase for uses to see how it is used.